### PR TITLE
Implement fixes proposed by Phoebe for ROOT v6.16.00.

### PR DESCRIPTION
**Pull request by yipengsun**
Part of umd-lhcb/root-curated#12.

### CODE REVIEW

File: roofit/histfactory/src/PiecewiseInterpolation.cxx [hunk 1]
```diff
@@ -219,6 +219,7 @@ void PiecewiseInterpolation::fillCacheObject(RooAbsCachedReal::FuncCacheElem& ca
     RooFIter paramIter(_paramSet.fwdIterator()) ;
 
     while((param=(RooAbsReal*)paramIter.next())) {
+      ++i;
```
Comment (CoffeeIntoScience) [inline roofit/histfactory/src/PiecewiseInterpolation.cxx:222]: if we increment here and `i` starts from zero we are always looking at the wrong `alpha`'s interp code

Comment (CoffeeIntoScience) [reply roofit/histfactory/src/PiecewiseInterpolation.cxx:222]: Please change line 215 to `int i=-1;`

Comment (CoffeeIntoScience) [reply roofit/histfactory/src/PiecewiseInterpolation.cxx:222]: The same changes are needed in `PiecewiseInterpolation::evaluate()` (same bug exists there). See lines 383 - 396

File: roofit/histfactory/src/PiecewiseInterpolation.cxx [hunk 2]
```diff
@@ -548,25 +548,9 @@ Bool_t PiecewiseInterpolation::setBinIntegrator(RooArgSet& allVars)
 }
 
 //_____________________________________________________________________________
-Int_t PiecewiseInterpolation::getAnalyticalIntegralWN(RooArgSet& allVars, RooArgSet& analVars,
```
Comment (CoffeeIntoScience) [inline roofit/histfactory/src/PiecewiseInterpolation.cxx:551]: This requires also removing the function prototype from the header file

**Review by CoffeeIntoScience — CHANGES_REQUESTED** [review]
See the inline comment. To recapitulate them here to maybe have things more compact

- moving the i increment up before you do the interpcode lookup means you need to start from -1 so your first lookup is i=0

- the evaluate method needs the same changes to the i counter

- removing the one method from this file requires removing the corresponding prototype from the header as well

**Comment by yipengsun** [comment]
Done! Also made sure that my editor is not removing trailing whitespaces so the diff is minimal.

**Review by CoffeeIntoScience — APPROVED** [review]
looks perfect!

**Pull request merged**